src/GeneratorState.hpp

```diff
@@ -0,0 +1,18 @@
+#ifndef POWER_WHISPERPOWER_GENERATORSTATE_HPP
+#define POWER_WHISPERPOWER_GENERATORSTATE_HPP
+
+namespace power_whisperpower {
+    /**
+     * @brief The state of the WhisperPower generator
+     *
+     */
+    enum GeneratorState {
+        RUNNING,
+        STOPPED,
+        STOPPING,
+        STARTING,
+        FAILURE
```

Comment:
Since orogen does not support enum class, it is needed to prefix the values with some discriminant ... e.g. GENSET_STATE_STOPPED, GENSET_STATE_...